src/util/command_line_handling.cpp

```diff
@@ -145,7 +145,8 @@ namespace hpx { namespace util
             std::size_t num_localities)
         {
             std::size_t batch_localities = env.retrieve_number_of_localities();
-            if (num_localities == 1)
+            if (num_localities == 1 &&
+                batch_localities != std::size_t(batch_localities))
```

Comment:
This doesn't make a lot of sense to me. Isn't this always false? `batch_localities` already is of type std::size_t.